src/prover_disk.hpp

```diff
@@ -110,10 +110,21 @@ class DiskProver {
 
         // The list of C2 entries is small enough to keep in memory. When proving, we can
         // read from disk the C1 and C3 entries.
+        uint64_t prev_c2_f7 = 0;
         auto* c2_buf = new uint8_t[c2_size];
         for (uint32_t i = 0; i < c2_entries - 1; i++) {
             SafeRead(disk_file, c2_buf, c2_size);
-            this->C2.push_back(Bits(c2_buf, c2_size, c2_size * 8).Slice(0, k).GetValue());
+
+            const uint64_t f7 = Bits(c2_buf, c2_size, c2_size * 8).Slice(0, k).GetValue();
+
+            // Short-circuit reading of the C2 table as soon as we encounter an f7 entry whose 
+            // value is lesser than the previous f7 read. This ensures that we don't read 
```

Comment:
Please do not leave trailing whitespace (here and on the previous line).

Reply:
Mmm looks like something got butchered when I copied the message from the wrong repo, and pasted it into the correct one. I'll fix it.